client/app/lib/components/redux-form/SingleFileInput/index.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { fieldMetaPropTypes } from 'redux-form';
+import { FormattedMessage, intlShape } from 'react-intl';
+import Dropzone from 'react-dropzone';
+import { grey300, red500 } from 'material-ui/styles/colors';
+import createComponent from '../createComponent';
+import mapError from '../mapError';
+import FilePreview from './FilePreview';
+import ImagePreview from './ImagePreview';
+import BadgePreview from './BadgePreview';
+
+const styles = {
+  dropzone: {
+    backgroundColor: grey300,
+    borderRadius: 5,
+    padding: 20,
+    textAlign: 'center',
+    width: '100%',
+  },
+  fileLabelError: {
+    color: red500,
+    display: 'inline-block',
+  },
+};
+
+/**
+ * Creates a Single file input component for use with Redux Forms.
+ * Available in two display modes: circular Avatar/Badge display (default) and non-circular image display.
+ * To use non-circular image display, pass in the prop `isNotBadge` with boolean value `true`.
+ *
+ * Additional format of form props (see createComponent for base set):
+ * {
+ *   ...createComponent,
+ *   value: {
+ *      url, // URL of preview of existing file if it is an image, otherwise nil.
+ *     name, // Name of existing file, if any.
+ *   },
+ * }
+ */
+// TODO: Use the input element as a controller component - https://reactjs.org/docs/forms.html
+class SingleFileInput extends React.Component {
+  static propTypes = {
+    meta: PropTypes.shape(fieldMetaPropTypes),
+    value: PropTypes.shape({
+      file: PropTypes.object,
+      url: PropTypes.string,
+      name: PropTypes.string,
+    }),
+    input: PropTypes.shape({
+      onChange: PropTypes.func.isRequired,
+    }),
+    errorMessage: PropTypes.string,
+    required: PropTypes.bool,
+    intl: intlShape.isRequired,
+    accept: PropTypes.string,
+    previewComponent: PropTypes.func,
```

Comment:
> The propTypes typechecking happens after defaultProps are resolved, so typechecking will also apply to the defaultProps

Shall we have this as `required`? Read this off the [React docs](https://reactjs.org/docs/typechecking-with-proptypes.html).

Reply:
We have a [lint rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/default-props-match-prop-types.md) in effect that requires the prop _not_ to be declared as required.

Reply:
>_< ok ignore this!